### PR TITLE
CMYK Support for SetColor functions

### PR DESCRIPTION
**Pull request by alphp**
Suppor for use CMYK colors in SetDrawColor(), SetFillColor() and SetTextColor().
If you pass no parameter, you get black.
If you pass 1 parameter, it is a grayscale component (between 0 and 100).
If you pass 3 parameters, they are RGB components (between 0 and 255).
If you pass 4 parameters, they are CMYK components (between 0 and 100).
Can you pass the color values as an array:
$rgb_color = [$red, $green, $blue];
$cmyk_color = [$cyan, $magenta, $yellow, $key]; // In CMYK the black color is called Key.

### CODE REVIEW

File: src/tFPDF/PDF.php [hunk 1]
```diff
@@ -892,13 +916,25 @@ public function SetFillColor($int_red, $int_green = null, $int_blue = null)
...
-            $this->str_text_color = sprintf('%.3F g', $int_red / 255);
-        } else {
-            $this->str_text_color = sprintf('%.3F %.3F %.3F rg', $int_red / 255, $int_green / 255, $int_blue / 255);
+        $args = func_get_args();
+        if (count($args) and is_array($args[0])) {
+            $args = $args[0];
+        }
+        switch (count($args)) {
+            case 1:
+                $this->str_text_color = sprintf('%.3f g', $args[0] / 100);
```
Comment (kazsaj) [inline src/tFPDF/PDF.php]: Your implementation of "g" divides argument value by 100, instead of the old 255. Is that on purpose or is this an oversight?
If it's on purpose do you have any reference links to help me understand why the change was necessary?

Comment (kazsaj) [inline src/tFPDF/PDF.php:928]: Your implementation of "g" divides argument value by 100, instead of the old 255. Is that on purpose or is this an oversight?
If it's on purpose do you have any reference links to help me understand why the change was necessary?

**Comment by kazsaj** [comment]
Sorry for taking so long to merge this change.

File: src/tFPDF/PDF.php (hunk 1 above)
Comment (alphp) [reply src/tFPDF/PDF.php:928]: Traditionally the RGB values have a range of 0 to 255, and the CMYK values of 0 to 100. The gray values follow the same range as the CMYK values, from 0 to 100.
This is true in most image and layout processing programs.

Comment (kazsaj) [reply src/tFPDF/PDF.php:928]: Yes, sorry I should have first read your PR description.
I did some testing and seems to work fine, hence the merge!

I will create a release with the updates included as soon as github stops having issues.

**Comment by kazsaj** [comment]
Thanks for your PR, this is now part of this repo in v2.2.0 release:
https://github.com/DocnetUK/tfpdf/releases/tag/v2.2.0
